### DISCUSSION TITLE
fix: Passing chat metadata to Pipe functions as an extra param

### DISCUSSION THREAD

**Discussion by DmitriyAlergant** [open-webui/open-webui#6666] [General]
Chat metadata (chat_id, message_id) can be very helpful inside functions for logging and tracking purposes.

However, currently it not available to functions (Pipe or Filter or Actions).  When calling functions, main.py passes over some extra_params (when such params match the function singnature) but "metadata" dict (that contains session_id, chat_id, message_id) is not in the list of passable parameters.

The PR: https://github.com/open-webui/open-webui/pull/6665 is a one-line change to fix this. I tested and it works. 
This is for the Pipe functions, I can also add the same fix or Filters and Actions.

The PR was closed for [tjbck](https://github.com/tjbck) without merging or comments, but perhaps with the help of this discussion it can be reinstated?

**Comment by DmitriyAlergant** [comment]
This was eventually accepted: https://github.com/open-webui/open-webui/commit/e3485d2

Closing the discussion

**Reply by VadimPoliakov** [reply]
And how to get in Pipelines?
